app/src/main/java/com/quran/labs/androidquran/database/TranslationsDBAdapter.java

```diff
@@ -72,23 +72,27 @@ public long getLastWriteTime() {
         null, null, null, null, null,
         TranslationsTable.ID + " ASC");
     if (cursor != null) {
-      while (cursor.moveToNext()) {
-        int id = cursor.getInt(0);
-        String name = cursor.getString(1);
-        String translator = cursor.getString(2);
-        String translatorForeign = cursor.getString(3);
-        String filename = cursor.getString(4);
-        String url = cursor.getString(5);
-        String languageCode = cursor.getString(6);
-        int version = cursor.getInt(7);
-        int minimumVersion = cursor.getInt(8);
-
-        if (quranFileUtils.hasTranslation(context, filename)) {
-          items.add(new LocalTranslation(id, filename, name, translator,
-              translatorForeign, url, languageCode, version, minimumVersion));
+      try {
```

Comment:
see comment below - try should move up though

Reply:
Salam, @ahmedre, can you explain why would we need to close a `null` cursor?

Reply:
wa3laikum alsalam - you don't, but you can't skip closing the cursor until you assign it either - so as a better pattern, you do something like:

```
Cursor cursor = null;
try {
  cursor = db.query()
  while (cursor.moveToNext()) {
  }
} finally {
  if (cursor != null) {
    cursor.close()
  }
}
```

`cursor.close()` can actually throw also, which is why we have the `DatabaseUtils.closeCursor` method which does that for you - it checks if it's null and if so does nothing - if it's not null, it tries to close it and swallows any exceptions it throws.

Reply:
Oh, ok, @ahmedre, you just shuffled the null check, lol.  I can't say that is necessarily "better"... actually it's a little more "traditional" e.g. to catch errors from cursor instantiation, in which case you'd need to keep a reference to it outside of try/catch.  But that's neither here nor there.

However, I'm curious about `DatabaseUtils.closeCursor`.  Why would we want to swallow `cursor.close()` and not `db.query()`.  I was going for consistency (ie throw everything)... but are you just wanting to ignore cursor close errors specifically?

Reply:
if `cursor.close()` fails, there's nothing i can do about it - there's nothing that can be fixed, etc so we swallow it. `db.query()` failing could be because of a bad query, missing column, etc and can fail for a number of reasons, and in all of these cases i want to know about it so i can fix it.